### PR TITLE
Update Env to add instance status filtering predicate

### PR DESCRIPTION
**Pull request by oswynb**
Requires: https://github.com/anchor/ceilometer-common/pull/10

Updates the Env supplied to ceilometer-common to include a predicate to filter instance statuses.

Makes minor deserialising changes ('/' -> '-')
groupMetrics no longer inserts an empty list into the Set of GroupMetrics it returns if none of the given metrics match anything in instances.

### CODE REVIEW

**Comment by fractalcat** [comment]
I think this needs a minor bump?

File: borel-core.cabal [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name:                borel-core
-version:             0.16.0
+version:             0.17.0
```
Comment (tranma) [inline borel-core.cabal]: I think we were going for 0.16.1

File: lib/Borel.hs [hunk 2]
```diff
@@ -80,8 +81,10 @@ groupMetrics :: Set Metric        -- ^ All available instances
              -> Set Metric        -- ^ Metrics to be grouped
              -> Set GroupedMetric
 groupMetrics instances metrics
-  =  let (allfs, nonfs) = _1 %~ S.toList $ S.partition (`S.member` instances) metrics
-     in  S.insert allfs $ S.map pure nonfs
+  =  let (allfs, nonfs) = (S.toList *** S.map pure) $ S.partition (`S.member` instances) metrics
+     in case allfs of
+       [] -> nonfs
+       xs -> S.insert xs nonfs
```
Comment (tranma) [inline lib/Borel.hs]: does this really make a difference? I intended for the identity case to cause an empty list of tags, and thus achieving the same effect.

File: lib/Borel/Ceilometer/Instance.hs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+billableStatus InstanceReboot           = True
+billableStatus InstanceHardReboot       = True
+billableStatus InstancePassword         = True
+billableStatus InstanceResize           = True
+billableStatus InstanceVerifyResize     = True
+billableStatus InstanceRevertResize     = True
+billableStatus InstancePaused           = False
+billableStatus InstanceSuspended        = False
+billableStatus InstanceRescue           = False
+billableStatus InstanceMigrating        = False
```
Comment (fractalcat) [inline lib/Borel/Ceilometer/Instance.hs]: Do we really not want to bill for migrating instances? It's not as if the customer has said "I have now stopped using this", and it's causing load.

Comment (oswynb) [inline lib/Borel/Ceilometer/Instance.hs]: Are the others OK? (did Bart get back to us on billing/not billing shut down instances?)

**Pull request merged**